### PR TITLE
Better match workflows

### PR DESCRIPTION
**Pull request by bendhouseart**
This fixes many of this issues that pop up when petdeface encounters different BIDS layouts, mostly:

1) anatomical file in each session folder
2) anatomical file in a single session folder
3) anatomical file in the subject directory (not in a session folder)

Will convert from draft when it's complete.

### CODE REVIEW

**Comment by bendhouseart** [comment]
Current status of Friday the :bat: 13th :bat:

```bash
$ for FOLDER in COX*; do tree $FOLDER/derivatives/petdeface; done
COX-1_analysis_faced_anat_in_each_session_folder/derivatives/petdeface
├── dataset_description.json
├── sub-PS50_ses-baseline
│   ├── anat
│   │   ├── sub-PS50_ses-baseline_T1w_defaced.nii
│   │   └── sub-PS50_ses-baseline_T1w_facemask.nii
│   └── pet
│       ├── registration.lta
│       └── sub-PS50_ses-baseline_pet_defaced.nii.gz
└── sub-PS50_ses-blocked
    ├── anat
    │   ├── sub-PS50_ses-blocked_T1w_defaced.nii
    │   └── sub-PS50_ses-blocked_T1w_facemask.nii
    └── pet
        ├── registration.lta
        └── sub-PS50_ses-blocked_pet_defaced.nii.gz

6 directories, 9 files
COX-1_analysis_faced_anat_in_first_session_folder/derivatives/petdeface
├── dataset_description.json
├── sub-PS50_ses-baseline
│   ├── anat
│   │   ├── sub-PS50_ses-baseline_T1w_defaced.nii
│   │   └── sub-PS50_ses-baseline_T1w_facemask.nii
│   └── pet
│       ├── registration.lta
│       └── sub-PS50_ses-baseline_pet_defaced.nii.gz
└── sub-PS50_ses-blocked
    └── pet
        ├── registration.lta
        └── sub-PS50_ses-blocked_pet_defaced.nii.gz

5 directories, 7 files
COX-1_analysis_faced_inherit_anat/derivatives/petdeface
├── dataset_description.json
├── sub-PS50
│   └── anat
│       ├── sub-PS50_T1w_defaced.nii
│       └── sub-PS50_T1w_facemask.nii
├── sub-PS50_ses-baseline
│   └── pet
│       ├── registration.lta
│       └── sub-PS50_ses-baseline_pet_defaced.nii.gz
└── sub-PS50_ses-blocked
    └── pet
        ├── registration.lta
        └── sub-PS50_ses-blocked_pet_defaced.nii.gz

6 directories, 7 files
```

**Comment by bendhouseart** [comment]
Pretty close to outputting proper bids directly to the derivatives directory, should be much easier now that I'm able to keep track of what workflow files belong to what inputs.

**Comment by bendhouseart** [comment]
@bilgelm tagging

File: petdeface/petdeface.py [hunk 1]
```diff
@@ -212,85 +220,82 @@ def init_single_subject_wf(subject_id: str, bids_dir: str) -> Workflow:
...
+        bids_data = BIDSLayout(bids_data)
+    elif isinstance(bids_data, BIDSLayout):
+        pass
 
-    if not subject_data["t1w"]:
-        raise RuntimeError(
-            "No T1w images found for participant {}. "
-            "All workflows require T1w images.".format(subject_id)
-        )
+    data = collect_anat_and_pet(bids_data)
```
Comment (bilgelm) [inline petdeface/petdeface.py:229]: Looks like the pet-mri matching is now being done by this petutils function. Cool!

**Review by bilgelm — APPROVED** [review]
The code looks a lot cleaner now.

**Review by mnoergaard — APPROVED** [review]
Overall LGTM.

**Pull request merged**